src/Sulu/Bundle/AdminBundle/Resources/js/components/DatePicker/DatePicker.js

```diff
@@ -102,6 +102,14 @@ export default class DatePicker extends React.Component<Props> {
 
     handleInputBlur = () => {
         if (this.inputChanged && typeof this.value === 'string') {
+            const fieldOptions = this.getFieldOptions(this.props.options);
+            const currentMoment = moment(this.value, this.getPlaceholder(fieldOptions));
+
+            if (currentMoment.isValid()) {
+                this.handleChange(currentMoment.toDate());
```

Comment:
Maybe one line:
`this.handleChange(currentMoment.isValid() ? currentMoment.toDate() : undefined);`?